src/rules/valid-expect.ts

```diff
@@ -90,6 +90,15 @@ const isAcceptableReturnNode = (
     AST_NODE_TYPES.AwaitExpression,
   ].includes(node.type);
 
+const isNoAssertionsParentNode = (node: TSESTree.Node): boolean => {
+  return (
+    node.type === AST_NODE_TYPES.ExpressionStatement ||
+    (node.type === AST_NODE_TYPES.AwaitExpression &&
+      node.parent != null &&
```

Comment:
`node.parent` can never be null, so you can just check for `undefined` (or better yet, just do `node.parent`)